### PR TITLE
[#5831] Fix handling of FullHttpResponse when respond to HEAD in Http…

### PR DESCRIPTION
**Pull request by normanmaurer**
…ServerCodec

Motivation:

cb139043f321836746915e7ada2d507ca45ed734 introduced special handling of response to HEAD requests. Due a bug we failed to handle FullHttpResponse correctly.

Modifications:

Correctly handle FullHttpResponse for HEAD requests.

Result:

Works as expected.

### CODE REVIEW

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java [hunk 1]
```diff
@@ -116,29 +110,41 @@ protected void encode(ChannelHandlerContext ctx, Object msg, List<Object> out) t
...
+
+                    break;
                 default:
                     throw new Error();
             }
+
+            if (msg instanceof LastHttpContent) {
+                state = ST_INIT;
+            }
         } else {
```
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java]: make this an `else if`

File: codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java [hunk 2]
```diff
@@ -184,8 +190,6 @@ private void encodeChunkedContent(ChannelHandlerContext ctx, Object msg, long co
                 buf.writeBytes(CRLF);
                 out.add(buf);
             }
-
-            state = ST_INIT;
         } else {
```
Comment (Scottmitch) [inline codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectEncoder.java]: make this an `else if`

**Comment by normanmaurer** [comment]
Merged into 4.1 (0eeeb764391490585f690c8b424d9ce2d43e6a9b) and 4.0 (0f069ae3119ffe7741df29f7e431b24786e5ac64)

**Comment by rkapsi** [comment]
@normanmaurer :+1: we just encountered it and can confirm this fixes it. ty!

**Comment by normanmaurer** [comment]
@rkapsi woooot :) thx

**Pull request closed without merging**